lib/examine-babelrc.js

```diff
@@ -0,0 +1,121 @@
+const verboseLog = require('./logger.js');
+const path = require('path');
+const fs = require('fs');
+const chalk = require('chalk');
+const process = require('process');
+const addSkipped = require('./rules-util.js').addSkipped;
+const processRules = require('./rules-util.js').processRules;
+const rulesConf = {
+    name: '.babelrc',
+    rules: [
+        {
+            desc: 'Should use plugin istanbul',
+            errorMsg:
+                'Add the following to babelrc:\n' +
+                '"env": {\n' +
+                '   "test": {\n' +
+                '       "plugins": [ "istanbul" ]\n' +
+                '   }\n' +
+                '}',
+            fn: usingIstanbulPlugin
+        },
+        {
+            desc: 'Should use babel-preset-env to polyfill browser support.',
+            errorMsg:
+                'Add the following to babelrc:\n' +
+                '"presets": [\n' +
+                '   ["babel-preset-env", {\n' +
+                '       "targets": {\n' +
+                '           "browsers": ["last 2 versions", "safari >= 7"]\n' +
+                '        }\n' +
+                '   }]\n' +
+                ']',
+            fn: usingBabelPresetEnv
+        }
+    ]
+};
+
+module.exports = function(options) {
+    'use strict';
+    const filenames = ['.babelrc', '.babelrc.json', '.babelrc.js'];
+    const abortPrefix = 'ABORTING babelrc examination';
+
+    // find which file name we can use since babelrc can be in many forms
+    const filteredFilenames = filenames.filter(filename => {
+        const filePath = path.join(process.cwd(), filename);
+        return fs.existsSync(filePath);
+    });
+
+    // didn't find any matching file name
+    if (!filteredFilenames.length) {
+        console.error(
+            chalk.red(`${abortPrefix}:`),
+            `No babel config found in ${process.cwd()}\n` +
+                'Please create a .babelrc file'
+        );
+        console.log(`Unable to process ${rulesConf.rules.length} rule(s)`);
+        return addSkipped(rulesConf);
+    }
+
+    const filename = filteredFilenames[0];
```

Comment:
If we always access the first entry of an array, maybe we should consider using filenames.reduce, or throw an error if multiple files are found for babel.